source/s3_client.c

```diff
@@ -1009,7 +1013,9 @@ static struct aws_s3_meta_request *s_s3_client_meta_request_factory_default(
 
                     client_max_part_size = g_s3_min_upload_part_size;
                 }
-                if (content_length <= client_part_size) {
+                uint64_t multipart_upload_threshold =
```

Comment:
corner case: what if multipart threshold is less than a part? should it fail or send 1 part?

Reply:
send one part of multipart upload as Java SDK did